nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/api/AbstractSession.java

```diff
@@ -2077,18 +2078,19 @@ public void setRetainUntil(DocumentRef docRef, Calendar retainUntil, String comm
         checkPermission(doc, SET_RETENTION);
         Map<String, Serializable> options = new HashMap<>();
         options.put(CoreEventConstants.RETAIN_UNTIL, retainUntil);
-        String commentStart = retainUntil == null ? null : retainUntil.toInstant().toString();
-        if (comment == null) {
-            comment = commentStart;
-        } else if (commentStart != null) {
-            comment = commentStart + " " + comment;
-        }
-        options.put("comment", comment);
+        boolean retainUntilIndeterminate = retainUntil == null
+                || RETAIN_UNTIL_INDETERMINATE.compareTo(retainUntil) == 0;
```

Comment:
Shouldn't we do the same thing than above in the code (ie: using Objects.equals)?

Reply:
I used `compareTo` instead of `equals` (Calendar.equals or Object.equals) because this last one (equals) uses the time zone to check equality: have look at `java.util.Calendar#equals` -> from my test  the declared  `RETAIN_UNTIL_INDETERMINATE` will have the UTC Time zone and the retrieved Calendar from `Calendar current = doc.getRetainUntil();` `current` will have `"Europe/Paris"` and the equality fails. I think before my fix and the adding test it's wasn't a problem because even the:

```
 if (Objects.equals(current, retainUntil)) {
            // unchanged, don't do anything
            return;
        }
```

will return false and the date are set for a second time even if the calendars are the same.


for more informations have a look at the exiting check: `org.nuxeo.ecm.core.storage.BaseDocument#allowNewRetention` 

and I don"t know how we deal with time zone in Nuxeo for Date type

@efge

Reply:
Ok,

Shouldn't we use compare in the equals then?

Reply:
Perhaps yes, I should confess that I didn't to want to change the  exiting code not related directly to this ticket

Reply:
Hum, I would say more or less because you don't use the same way to compare the dates. I would say we should use the same way to compare them or if there's a reason, let's put it in a comment, wdyt?

Reply:
to do the correct thing, let aligne the `equals` to use the compare like i did and like it's done on low level. I think i will use the same commit, i mean it's really a minor change to separate it into a specific commit wdyt ?

Reply:
I think you can use the same commit, it would make sense to have it next to `compare` introduction you're doing 👍.